src/test/java/io/zeebe/containers/cluster/ZeebeClusterBuilderTest.java

```diff
@@ -509,6 +509,319 @@ void shouldNotConfigureEmbeddedGateway() {
         .doesNotContainEntry("ZEEBE_BROKER_GATEWAY_ENABLE", "true");
   }
 
+  @Test
+  void shouldApplyBrokerConfigurationOnlyOnBrokers() {
+    // given
+    final String foreseeEnv = "IS_CONFIGURED_BY_BROKER_FUNCTION";
+    final ZeebeClusterBuilder builder =
+        new ZeebeClusterBuilder()
+            .withBrokerConfig(broker -> broker.addEnv(foreseeEnv, ""))
+            .withBrokersCount(1)
+            .withGatewaysCount(1)
+            .withEmbeddedGateway(false);
+
+    // when
+    final ZeebeCluster cluster = builder.build();
+
+    // then
+    assertThat(cluster.getBrokers())
```

Comment:
:wrench: As you only care to assert on the values, you can already here pass it to the assert builder, e.g. `assertThat(cluster.getBrokers().values())`. Additionally, you probably don't need to outer `as` since the inner assert has a good error message, but you can also keep it, it doesn't hurt ;) To see, try failing the test with and without it, and then imagine that you're a new developer on this and you're trying to figure out what it's asserting/what went wrong during failure, and see what helps you best.

Reply:
The suggestion applies to the assertions below as well for nodes and gateways, and in the tests below with similar assertions :slightly_smiling_face:

Reply:
Done